src/main/kotlin/apply/application/EvaluationTargetService.kt

```diff
@@ -0,0 +1,147 @@
+import org.springframework.stereotype.Service
+import javax.annotation.PostConstruct
+import javax.transaction.Transactional
+
+@Transactional
+@Service
+class EvaluationTargetService(
+    private val evaluationTargetRepository: EvaluationTargetRepository,
+    private val evaluationRepository: EvaluationRepository,
+    private val applicationFormRepository: ApplicationFormRepository,
+    private val applicantRepository: ApplicantRepository,
+    private val cheaterRepository: CheaterRepository
+) {
+    fun findByEvaluationId(evaluationId: Long): List<EvaluationTarget> =
+        evaluationTargetRepository.findByEvaluationId(evaluationId)
+
+    fun update(evaluationId: Long) {
+        val evaluation = evaluationRepository.findByIdOrNull(evaluationId) ?: throw IllegalArgumentException()
+        val cachedCheaterApplicantIds = cheaterRepository.findAll().map { it.applicantId }
+
+        return when {
+            !evaluation.hasBeforeEvaluation() -> updateEvaluationTargetOfFirst(evaluation, cachedCheaterApplicantIds)
+            evaluation.hasBeforeEvaluation() -> updateEvaluationTarget(evaluation, cachedCheaterApplicantIds)
+            else -> throw IllegalStateException("평가 대상자를 갱신할 수 없습니다.")
+        }
+    }
+
+    private fun updateEvaluationTargetOfFirst(evaluation: Evaluation, cachedCheaterApplicantIds: List<Long>) {
+        val oldApplicantIds = evaluationTargetRepository.findByEvaluationId(evaluation.id)
+            .map { it.applicantId }
+            .toSet()
+
+        val newApplicantIds = createEvaluationTargetsFromFirst(evaluation)
+            .map { it.applicantId }
+            .filter { !cachedCheaterApplicantIds.contains(it) }
+            .toSet()
+
+        val cheaterApplicantIds = (oldApplicantIds + newApplicantIds).filter { cachedCheaterApplicantIds.contains(it) }
+
+        if (isAlreadyUpToDate(oldApplicantIds, cheaterApplicantIds, newApplicantIds)) {
+            delete(evaluation, cheaterApplicantIds)
+        } else {
+            deleteAndSave(oldApplicantIds, newApplicantIds, cheaterApplicantIds, evaluation)
+        }
+    }
+
+    private fun createEvaluationTargetsFromFirst(evaluation: Evaluation): List<EvaluationTarget> {
+        val applicantIds = applicationFormRepository.findByRecruitmentId(evaluation.recruitmentId)
+            .map { it.applicantId }
+
+        return applicantRepository.findAllById(applicantIds)
+            .map { EvaluationTarget(evaluationId = evaluation.id, applicantId = it.id) }
+    }
+
+    private fun isAlreadyUpToDate(
+        oldApplicantIds: Set<Long>,
+        cheaterApplicantIds: List<Long>,
+        newApplicantIds: Set<Long>
+    ): Boolean {
+        val oldNormalApplicantIds = oldApplicantIds - cheaterApplicantIds
+        val newNormalApplicantIds = newApplicantIds - cheaterApplicantIds
+
+        return oldNormalApplicantIds.containsAll(newNormalApplicantIds) &&
+            newNormalApplicantIds.containsAll(oldNormalApplicantIds)
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        return oldNormalApplicantIds == newNormalApplicantIds
```

Reply:
이 부분을 학습테스트로 처리해봤을 때에 돌아가지 않아서 저렇게 구현했었는데, 제이슨이 피드백 남겨주신 부분을 확인하고 다시 테스트를 해보니 돌아가는군요😇

==를 이용해서 비교하도록 하겠습니다.

Reply:
`List`와 `Set`의 차이일 것 같아요.

Reply:
이번 pr의 Enum Equals와 == 비교부터 지금까지를 보아하니, 코틀린의 기본을 깊게 공부해야겠습니다!

Reply:
위의 동작들은 Java에도 적용됩니다. 참고로 코틀린의 `==`는 Java의 `equals`와 동일하고 코틀린의 `===`는 Java의 `==`와 동일합니다.